reportssql/treatment_report_functions.sql

```diff
@@ -168,6 +168,38 @@ WHERE
 END$$ 
 DELIMITER ;
 
+DROP FUNCTION IF EXISTS TREATMENT_Indicator4b;
+
+DELIMITER $$
+CREATE FUNCTION TREATMENT_Indicator4b(
+    p_startDate DATE,
+    p_endDate DATE,
+    p_startAge INT(11),
+    p_endAge INT (11),
+    p_includeEndAge TINYINT(1),
+    p_gender VARCHAR(1)) RETURNS INT(11)
+    DETERMINISTIC
+BEGIN
+    DECLARE result INT(11) DEFAULT 0;
+
+SELECT
+    COUNT(DISTINCT pat.patient_id) INTO result
+FROM
+    patient pat
+WHERE
+    patientGenderIs(pat.patient_id, p_gender) AND
+    patientAgeWhenRegisteredForHivProgramIsBetween(pat.patient_id, p_startAge, p_endAge, p_includeEndAge) AND
+    patientHasEnrolledIntoHivProgram(pat.patient_id) = "Yes" AND
+    patientHasTherapeuticLine(pat.patient_id, 0) AND
+    patientHasStartedARVTreatmentBefore(pat.patient_id, p_endDate) AND
+    patientWasOnARVTreatmentByDate(pat.patient_id, p_startDate) AND
+    patientIsLostToFollowUp(pat.patient_id, p_startDate, p_endDate) AND
```

Comment:
@tmvumbi2 I chatted to Richard, and for this report we should check for current date and not p_endDate in the patientIsLostToFollowUp function. Does it make sense? (see test case 4)